src/foam/nanos/auth/ServiceProviderAwareDAO.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'ServiceProviderAwareDAO',
+  extends: 'foam.nanos.auth.AuthorizationDAO',
```

Comment:
This doesn't seem right to me. A decorator that adds awareness of the SPID property shouldn't have anything to do with authorization checks.